docs/superpowers/specs/2026-05-03-genai-features-batch-design.md

```
@@ -0,0 +1,365 @@
+Infrastructure already exists:
+- `llm_cache` table with columns: id, prompt, response, system_message, embedding, hit_count, last_hit, created_at
+- `llm_cache_vec` virtual table (sqlite-vec `vec0`) with embedding float[1536]
+- `LLM_Bridge::get_text_embedding()` — generates embeddings via Ollama
+- Config: `genai_llm_cache_similarity_threshold` (default 85) and `genai_llm_cache_enabled` (default true)
+- Stats: atomic counters for cache hits/misses/lookups/stores/timing in `AI_Features_Manager`
+
+### Design
+
+#### `check_cache()`
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)